sale_commission/wizard/wizard_settle.py

```diff
@@ -85,11 +85,15 @@ def action_settle(self):
                         settlement = settlement_obj.create(
                             {'agent': agent.id,
                              'date_from': sett_from,
-                             'date_to': sett_to})
+                             'date_to': sett_to,
+                             'company_id':
+                                agent_lines[pos].invoice_line.company_id.id})
```

Comment:
This doesn't serve. You can have accessed to more than one company with your user, so your invoices will be mixed. You have to group invoices by company, and make one settlement per group.